e2e/plugin-e2e/various-suite/mysql.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { selectors } from '@grafana/e2e-selectors';
+import { expect, test } from '@grafana/plugin-e2e';
+
+import {
+  tablesResponse,
+  fieldsResponse,
+  datasetResponse,
+  normalTableName,
+  tableNameWithSpecialCharacter,
+} from './mocks/mysql.mocks';
+
+test.beforeEach(async ({ context, selectors, explorePage }) => {
+  await explorePage.datasource.set('gdev-mysql');
+  await context.route(selectors.apis.DataSource.queryPattern, async (route, request) => {
+    switch (request.postDataJSON().queries[0].refId) {
+      case 'tables':
+        return route.fulfill({ json: tablesResponse, status: 200 });
+      case 'fields':
+        return route.fulfill({ json: fieldsResponse, status: 200 });
+      case 'datasets':
+        return route.fulfill({ json: datasetResponse, status: 200 });
+      default:
+        return route.continue();
+    }
+  });
+});
+
+test('code editor autocomplete should handle table name escaping/quoting', async ({ explorePage, selectors, page }) => {
+  await page.getByLabel('Code').check();
+  const editor = explorePage.getByTestIdOrAriaLabel(selectors.components.CodeEditor.container).getByRole('textbox');
```

Comment:
Didn't know you could do this! I wonder if it's a bit brittle to rely on monaco internal elements though. If for example they decide to change element from textbox to textarea, this would no longer work. Maybe safer to do something like this:
```ts
await page.waitForFunction(() => (window as any).monaco);
await explorePage.getByTestIdOrAriaLabel(selectors.components.CodeEditor.container).click();
await page.keyboard.insertText('S');
```

Then the test interacts with the UI exactly the same way as a real user would.

Reply:
this is locating the element which has the role textbox not an input or textarea. I tried to use what you suggested but checking the value in line 36 would require to get the textbox element anyway.